server/src/main/java/io/druid/server/http/DatasourcesResource.java

```diff
@@ -180,15 +179,28 @@ public Response deleteDataSource(
     if (indexingServiceClient == null) {
       return Response.ok(ImmutableMap.of("error", "no indexing service found")).build();
     }
+
     if (kill != null && Boolean.valueOf(kill)) {
       try {
         indexingServiceClient.killSegments(dataSourceName, new Interval(interval));
       }
-      catch (Exception e) {
+      catch (IllegalArgumentException e) {
+        return Response.serverError().entity(
+            ImmutableMap.of(
+                "error",
+                "Exception occurred. Probably the interval is invalid",
+                "Error-Message",
+                e
+            )
+        )
+                       .build();
+      } catch (Exception e) {
         return Response.serverError().entity(
             ImmutableMap.of(
                 "error",
-                "Exception occurred. Are you sure you have an indexing service?"
+                "Exception occurred. Are you sure you have an indexing service?",
+                "Error-Message",
```

Comment:
yes but I wanted to have `Exception` class name to be part of the `Error-Message`. If I use `e.getMessage()` it does not include the Exception class